codecs/src/per/common/decode/mod.rs

```diff
@@ -110,6 +112,64 @@ pub fn decode_integer_common(
     Ok((value, extended_value))
 }
 
+// Common function to decode a REAL value
```

Comment:
A General comment, if it is possible to refer to section numbers in the standard, it would be a good idea where each function comes from. If it's not possible, we can just refer to the section for real values encoding in X.690.

Also: do you want to add `log::trace` at certain points. This should help in debugging if and when we find issues with `(en|de)coding` the values.

Reply:
Sure, I should be able to add these comments in. I will also see if it makes sense to sprinkle in some `log::trace` statements.

Reply:
I have added some comments and trace statements. I might have gone a bit overboard in the comments in the decoder. Does this look better?

Reply:
I generally add `log::trace` to entry of the function. Often that is good enough! Going overboard never hurts. I believe I need to review logging in codec in general (see some places two `trace` outputs that are identical. 

Also, with documenting references, the more is always better! :-) 

This is perfect for now!